### PR TITLE
[CINN] Removed 0d-to-1d pass

### PR DESCRIPTION
**Pull request by lshpku**
### PR Category
CINN

### PR Types
Improvements

### Description
Removed the Convert0DTo1DPass.

### CODE REVIEW

File: paddle/cinn/operator_fusion/utils.h [hunk 1]
```diff
@@ -50,6 +50,10 @@ static std::vector<int64_t> GetReduceAxisIdx(pir::Operation* reduce_op) {
   CHECK(attr_val.isa<::pir::ArrayAttribute>());
   const auto& axis_attr = attr_val.dyn_cast<::pir::ArrayAttribute>();
   std::vector<int64_t> reduce_axis_idx;
+  if (input_rank == 0) {
+    VLOG(4) << "GetReduceAxisIdx: ";
```
Comment (zyfncg) [inline paddle/cinn/operator_fusion/utils.h]: LOG信息可以描述的更准确些

File: test/ir/pir/cinn/symbolic/test_cinn_0d_tensor.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
+# Copyright (c) 2023 PaddlePaddle Authors. All Rights Reserved.
```
Comment (zyfncg) [inline test/ir/pir/cinn/symbolic/test_cinn_0d_tensor.py]: 2023 -> 2024

**Review by tc20042008 — DISMISSED** [review]

File: paddle/cinn/operator_fusion/utils.h (hunk 1 above)
Comment (jiahy0825) [inline paddle/cinn/operator_fusion/utils.h]:
```suggestion
    VLOG(4) << "Reduce op has 0D Tensor input, return empty reduce_axis";
```

File: test/ir/pir/cinn/symbolic/test_cinn_0d_tensor.py (hunk 2 above)
Comment (jiahy0825) [inline test/ir/pir/cinn/symbolic/test_cinn_0d_tensor.py]:
```suggestion
# Copyright (c) 2024 PaddlePaddle Authors. All Rights Reserved.
```

**Review by zyfncg — APPROVED** [review]

**Review by 2742195759 — APPROVED** [review]
LGTM

**Pull request merged**